### ISSUE TITLE
About BitSrunLogin/encryption/srun_base64.py

### ISSUE THREAD

**Issue by saber65535** [coffeehat/BIT-srun-login-script#1]
Hi,
I have seen your share at Zhihu. And your work is really wonderful. There is a little bug in your code.
First I run your code in my NAS system. It doesn't works. And then I use my Mac and another NAS to run this code. All works.
The wrong info is under below.
```
Step3: Loggin and resolve response.
{"password":"123","enc_ver":"srun_bx1","username":"123","acid":"1","ip":"10.114.111.182"}
Traceback (most recent call last):
  File "demo.py", line 8, in <module>
    password = "123"
  File "/volume1/NAS/NAS-Workdir/srun/BitSrunLogin/LoginManager.py", line 44, in login
    self.get_login_responce()
  File "/volume1/NAS/NAS-Workdir/srun/BitSrunLogin/LoginManager.py", line 61, in get_login_responce
    self._generate_encrypted_login_info()
  File "/volume1/NAS/NAS-Workdir/srun/BitSrunLogin/LoginManager.py", line 170, in _generate_encrypted_login_info
    self._encrypt_info()
  File "/volume1/NAS/NAS-Workdir/srun/BitSrunLogin/_decorators.py", line 14, in wrapper
    return func(self, *args, **kwargs)
  File "/volume1/NAS/NAS-Workdir/srun/BitSrunLogin/_decorators.py", line 14, in wrapper
    return func(self, *args, **kwargs)
  File "/volume1/NAS/NAS-Workdir/srun/BitSrunLogin/LoginManager.py", line 142, in _encrypt_info
    self.encrypted_info = "{SRBX1}" + get_base64(get_xencode(self.info, self.token))
  File "/volume1/NAS/NAS-Workdir/srun/BitSrunLogin/encryption/srun_base64.py", line 32, in get_base64
    b10 = (_getbyte(s, i) << 16) | (_getbyte(s, i + 1) << 8);
  File "/volume1/NAS/NAS-Workdir/srun/BitSrunLogin/encryption/srun_base64.py", line 7, in _getbyte
    x = ord(s[i])
IndexError: string index out of range
```
And then I find out your reference link at [CSDN](https://blog.csdn.net/qq_41797946/article/details/89417722) .  One User called zzdyyy mentioned that, in `get_base64` last `else` should be replace with `elif len(s) - imax ==2`. Because when the length of string which post to `get_base64` is exactly a multiple of 3, use `_getbyte` would export Error just like above.

Nice day

**Comment by coffeehat** [comment]
Thank you for finding this bug o(*￣▽￣*)ブ, 
I'll fix it soon

**Comment by coffeehat** [comment]
Sorry for the late patch
The bug is confirmed. When logining, if the base64 of your encrypted user info happens to be multiples of three, this bug will emerge. Someone on Zhihu also encounter this problem.
The bug is fixed in commit #5064b2b. You can test it when convenient.
Thank you for the feedback

**Comment by coffeehat** [comment]
This issue will be closed if it works on your environment
